google-cloud-spanner/src/main/java/com/google/cloud/spanner/Value.java

```diff
@@ -218,7 +224,82 @@ public static Value pgJsonb(@Nullable String v) {
   }
 
   /**
-   * Returns a {@code BYTES} value.
+   * Return a {@code PROTO} value.
+   *
+   * @param v Not null Proto message.
+   */
+  public static Value protoMessage(AbstractMessage v) {
+    Preconditions.checkNotNull(v, "Use protoMessage(ByteArray, Descriptor) for null values.");
+    return protoMessage(
+        ByteArray.copyFrom(v.toByteArray()), v.getDescriptorForType().getFullName());
+  }
+
+  /**
+   * Return a {@code PROTO} value
+   *
+   * @param v Serialized Proto Array, which may be null.
+   * @param protoTypFqn Fully qualified name of proto representing the proto definition.
+   */
+  public static Value protoMessage(@Nullable ByteArray v, String protoTypFqn) {
+    return new ProtoMessageImpl(v == null, v, protoTypFqn);
+  }
+
+  /**
+   * Return a {@code PROTO} value
+   *
+   * @param v Serialized Proto Array, which may be null.
+   * @param descriptor Proto Type Descriptor, use static method {@code
+   *     MESSAGE.getDescriptorForType()}.
+   */
+  public static Value protoMessage(@Nullable ByteArray v, Descriptor descriptor) {
+    Preconditions.checkNotNull(descriptor, "descriptor can't be null.");
+    return new ProtoMessageImpl(v == null, v, descriptor.getFullName());
+  }
+
+  /**
+   * Return a {@code ENUM} value.
+   *
+   * @param v Proto Enum, which may be null.
+   */
+  public static Value protoEnum(ProtocolMessageEnum v) {
+    Preconditions.checkNotNull(v, "Use protoEnum(Long, Descriptor) for null values.");
+    return protoEnum(v.getNumber(), v.getDescriptorForType().getFullName());
+  }
+
+  /**
+   * Return a {@code ENUM} value.
+   *
+   * @param v Enum non-primitive Integer constant.
```

Comment:
This should explain more what the value should and/or is allowed be. Is 100 ok? And why is the argument type `Long`, if it should be an `Integer`?

Reply:
1. Yes, user can pass the 100 but backend will throw the exception if there is no corresponding Enum value for Integer Constant. We can add a comment to make it clear. 
2.  We used Long to provide compatibility between  Int64 and Enum.